### PR TITLE
Split secret tree and encryption tests

### LINKED ISSUE #67: Split secret tree tests from encryption test vector
Right now, the encryption test vectors cover both the derivation of encryption keys and the encrypted message framing.  It would be cleaner and have more coverage to split this into a few smaller cases:

* A "secret tree" test focused on verifying that the secret tree derivation and hash ratchets are implemented correctly
* A signing test that verifies the FramedContent <-> PublicMessage transformation
* An encryption test that verifies the FramedContent <-> PrivateMessage transformation

For suites for whom signing is non-deterministic, the signing test would be a verification test only, and the encryption test might have to be unprotect-only or look at AuthenticatedContent <-> PrivateMessage.  But these could be much smaller and faster, using single sets of keys as opposed to `n_leaves * n_generations` useless repetitions.

### PR DESCRIPTION
**Pull request by bifurcation**
Fixes #67

### CODE REVIEW

File: test-vectors.md [hunk 1]
```diff
@@ -71,81 +71,115 @@ Verification:
...
+## Message Protection
+
+Parameters:
+* Ciphersuite
+
+Format:
+
+``` text
+{
+  "cipher_suite": /* uint16 */,
```
Comment (mulmarta) [inline test-vectors.md:136]: Is the sender index missing from this struct?

File: test-vectors.md [hunk 2]
```diff
@@ -71,81 +71,115 @@ Verification:
...
+  "proposal":  /* serialized Proposal */,
+  "proposal_pub":  /* serialized MLSMessage(PublicMessage) */,
+  "proposal_priv":  /* serialized MLSMessage(PrivateMessage) */,
+
+  "commit":  /* serialized Commit */,
+  "commit_pub":  /* serialized MLSMessage(PublicMessage) */,
+  "commit_priv":  /* serialized MLSMessage(PrivateMessage) */,
+
+  "application":  /* hex-encoded binary application data */,
+  "application_pub":  /* serialized MLSMessage(PublicMessage) */,
```
[the remark this answers is not in the document]
Comment (bifurcation) [reply test-vectors.md:159]: Oh good point.

```suggestion
```

File: test-vectors.md (hunk 1 above)
Comment (mulmarta) [reply test-vectors.md:136]: My reason for including `sender` was much more benign -- it's simply useful to know the sender in advance s.t. we can set "myself" to not be the sender (our decryption function throws an error in such case).

Re. known-answer tests, let's not do it here, the test case is long enough as it is. FWIW, the way we do it is to have a "set_reuse_guard" function which is only compiled in tests (else it's internally chosen at random). Seems safe enough

Comment (bifurcation) [reply test-vectors.md:136]: Yeah, I had some similar thoughts while implementing yesterday.  We could do something simple: Just have `n_leaves` always be 2, and always send from sender 1 (say).  That way, we can eliminate `n_leaves` from the test vector and avoid the issue you note.

Comment (mulmarta) [reply test-vectors.md:217]: Sounds good!

**Pull request merged**